src/TCP.cxx

```diff
@@ -0,0 +1,79 @@
+// MIT License
+//
+// Copyright (c) 2021 Felix Moessbauer
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+///
+/// \author Felix Moessbauer
+///
+
+#include "TCP.h"
+#include "InfluxDBException.h"
+#include <string>
+
+namespace influxdb::transports
+{
+
+    TCP::TCP(const std::string& hostname, int port)
+        : mSocket(mIoService, boost::asio::ip::tcp::endpoint(boost::asio::ip::tcp::v4(), 0))
+    {
+        boost::asio::ip::tcp::resolver resolver(mIoService);
+        boost::asio::ip::tcp::resolver::query query(boost::asio::ip::tcp::v4(), hostname, std::to_string(port));
```

Comment:
Just saw that this is ipv4 only. For our use-case back then that was ok, but IMHO we should support both.